extension/src/content/content.js

```diff
@@ -87,36 +80,72 @@ window.addEventListener("load", () => {
         fetchState === fetchStates.idle &&
         // there is more than 10 characters in the textarea
-        wiseTextarea.innerHTML.length > 10
+        textarea.value.length > 10
       ) {
         controller.abort();
         controller = new AbortController();
 
         fetchState = fetchStates.fetching;
-        debuggingLogger("before fetchAutocomplete event");
-        const fetchedAutocompleteText = await fetchAutocomplete({
+
+        const { body, aborted, error } = await fetchAutocomplete({
           textarea,
           controller,
         });
-        debuggingLogger("after fetchAutocomplete event");
-        if (fetchState === fetchStates.idle) {
-          // do nothing with the response
+        console.log("body", body, "aborted", aborted, "error", error);
+        if (aborted) {
+          console.log("aborted!!");
+          return;
+        } else if (error) {
+          console.log("error!!");
+          console.error(error);
           return;
         }
-        if (fetchedAutocompleteText) {
-          fetchState = fetchStates.fetched;
-          let newAutocompleteText = fetchedAutocompleteText;
-
-          if (textarea.textContent.slice(-1) !== " ") {
-            newAutocompleteText = " " + newAutocompleteText;
+        const reader = body.getReader();
+        const processStream = async () => {
+          while (true) {
+            const { value, done } = await reader.read();
+            if (done) {
+              break;
+            }
+
+            const chunkString = new TextDecoder("utf-8").decode(value);
+
+            const dataChunks = chunkString.split("data: ");
+            const nonEmptyChunks = dataChunks.filter(
+              (chunk) => chunk.trim() !== ""
+            );
+            const jsonStrings = nonEmptyChunks.map((chunk) => {
+              const trimmedChunk = chunk.trim();
+              if (trimmedChunk === "[DONE]") {
+                return null; // Skip further processing for [DONE] case
+              }
+              return trimmedChunk.replace(/^data:\s*/, "");
+            });
+            const validJsonStrings = jsonStrings.filter(
+              (jsonString) => jsonString !== null
+            );
+
+            const content = validJsonStrings
+              .map(
+                (jsonObject) => JSON.parse(jsonObject).choices[0].delta.content
+              )
+              .join("");
+
+            autocompleteText.textContent +=
+              !autocompleteText.textContent.length &&
+              textarea.value.slice(-1) !== " "
+                ? " " + content
+                : content;
           }
-          autocompleteText.textContent = newAutocompleteText;
-
-          textarea.classList.add("expanded-textarea");
-          wiseTextarea.classList.add("expanded-textarea");
-        } else {
-          fetchState = fetchStates.error;
+        };
+        if (fetchState === fetchStates.idle) {
+          // do nothing with the response
+          return;
```

Comment:
handles response if fetch state is back in idle (I think this is for cancelled fetches)